ts/private/ts_project_worker.js

```diff
@@ -771,12 +767,18 @@ function createProgram(args, inputs, output, exit) {
     }
 
     function createDirectory(p) {
-        p = p.replace(SYNTHETIC_OUTDIR, ".")
+        p = p.replace(SYNTHETIC_OUTDIR, "")
         ts.sys.createDirectory(p);
     }
 
     function writeFile(p, data, mark) {
-        p = p.replace(SYNTHETIC_OUTDIR, ".")
+        p = p.replace(SYNTHETIC_OUTDIR, "")
+        if (p.endsWith(".map")) {
+            // We need to postprocess map files to fix paths for the sources. This is required because we have a SYNTHETIC_OUTDIR suffix and 
+            // tsc tries to relativitize sources back to rootDir. in order to fix it the leading `../` needed to be stripped out.
+            // TODO: figure out how to make tsc do this for us.
```

Comment:
Can you expand on this comment a bit? I think the TODO should drop since we know its not possible with tsc in the current impl. Better to replace it with a more detailed explanation of the problem and what we tried that didn't work (namely mapRoot & the `../path/file` approach which broke in the root package.

Reply:
Sure.